client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -197,25 +197,24 @@ private synchronized PlatformError startTunnel(
       }
     }
 
-    final NewClientResult clientResult = Outline.newClient(config.transportConfig);
-    if (clientResult.getError() != null) {
-      LOG.log(Level.WARNING, "Failed to create Outline Client", clientResult.getError());
+    final NewRemoteDeviceResult deviceResult = Outline.newRemoteDevice(config.transportConfig);
+    if (deviceResult.getError() != null) {
+      LOG.log(Level.WARNING, "Failed to create Outline remote IP device", deviceResult.getError());
       tearDownActiveTunnel();
-      return clientResult.getError();
+      return deviceResult.getError();
     }
-    final outline.Client client = clientResult.getClient();
+    final outline.RemoteDevice remoteDevice = deviceResult.getRemoteDevice();
 
-    PlatformError udpConnError = null;
+    PlatformError deviceConnError = null;
     if (!isAutoStart) {
       try {
         // Do not perform connectivity checks when connecting on startup. We should avoid failing
         // the connection due to a network error, as network may not be ready.
-        final TCPAndUDPConnectivityResult connResult = checkServerConnectivity(client);
-        if (connResult.getTCPError() != null) {
+        deviceConnError = remoteDevice.HasConnectivity();
```

Comment:
Probably `TestConnectivity` would be a better name. `HasConnectivity` sounds like it will return a `bool` instead of an `Error`.

Reply:
Sounds good.